### PR TITLE
Ticket 7, Delete a Post

### PR DESCRIPTION
**Pull request by nthi**
# Description
[Trello ticket 7](https://trello.com/c/cssEp65s/7-7-delete-a-post)

This allows a user to delete a post they have written. If they select "Delete" from any view with delete, they will be presented with a delete confirmation view.

If they confirm delete, the post will be removed from system and user will be directed back to Post list.

If they do not confirm delete (by clicking "Return to post" link instead of "DELETE" button), they will be directed back to the post details.

Files changed:
PostRepository.cs
(Adds DeletePost(int id) method)

IPostRepository.cs
(Adds DeletePost(int id) to the interfaces)

PostController.cs
(Adds public IActionResult Delete(int id) (GET)
Adds public IActionResult Delete(int id, Post post) (POST))

Creates file:
Views > Post > Delete.cshtml
(Creates Delete view. form asp-action at end of Delete view includes Delete button and "Back to post" @Html.ActionLink)

# Testing Instructions
1. git add and git commit whatever you're working on on your branch
2. git checkout main
3. git fetch --a (or git fetch --all)
4. git checkout nt-deletePost-ticket-7
5. If you don't see the most recent version of my files, git pull origin nt-deletePost-ticket-7
6. If you haven't created the Tabloid MVC database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/01_TabloidMVC_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidmvc-the-caramel-apple-pops/blob/main/SQL/02_TabloidMVC_Seed_Data.sql)
9. Click the execute with debugger solid green button to launch the app
10. In browser, please go to Login on the nav bar and log in with this email: admin@example.com
11. If you are already logged in as admin/Once you are logged in as admin, click the "Posts" link in the nav bar. This takes you to a list of all posts. If you do not have many posts, you should create a couple more so you will have extras to delete.
12. Choose a post and click the "Delete" option at the end of the row. This takes you to a Delete confirmation page with full post details, a Delete button, and a Return to Post link.
13. Click Delete. You should be returned to the Posts list.
14. Choose a post from the list and click the "View" option. When you are in a detailed view of the post, you should see the option to Delete.
15. Click Delete. You should see the same sort of Delete confirmation page as before. 
16. Click the "Back to Post" link. You should be returned to the detailed view of the post.
17. Repeat these steps (12-16) from the "My Posts" list view.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by jeannettegeib — APPROVED** [review]
I tested your branch and this feature works as described. Huzzah!  :-)
-Jeannette

**Comment by sarahebrooks12** [comment]
Instructions look good. Please have two testing approvals before merging

**Review by julslt34 — APPROVED** [review]
I walked through adding a post in both post and my post pages and the code worked as instructed.  Great Job!

**Pull request merged**
